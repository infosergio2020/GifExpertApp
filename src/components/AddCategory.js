import React,{ useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        setInputValue( e.target.value );
        console.log('handleInputChange llamado');
    }

    const handleSubmit = (e)=>{
        console.log('hadleSubmit',inputValue);
        e.preventDefault();
        
        if(inputValue.trim().length > 2 ){
            setCategories( cats => [inputValue,...cats]);
            setInputValue('');
        }else {
            // alert('Debe tener mas de 2 caracteres');
            setInputValue('');
        }
        
    }

    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input className="input input-background" 
                   type="text"
                   value={inputValue}
                   onChange={handleInputChange}>
            </input>
        </form>
    )
}

// con esto prevengo que alguien que reutilize mi funcion se olvide de mandar una funcion como parametro
AddCategory.propTypes = { 
    setCategories: PropTypes.func.isRequired,
}