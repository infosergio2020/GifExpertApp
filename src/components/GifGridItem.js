import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ( {title,url} ) => {
    return (
        <div className="card animate__animated animate__flipInX">
            <img src={url} alt={title}></img>
            <p> {title} </p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired, //indico que solo acepta este tipo
    url: PropTypes.string.isRequired
}


/*
    1. Enzyme
    2. Enzyme to Json
    3. Debe de mostrar el componente correctamente
        * shallow
        * wrapper
        * wrapper .toMatchSnapshot()
*/