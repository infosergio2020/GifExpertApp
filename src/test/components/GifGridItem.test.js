import React from "react";
import { shallow } from 'enzyme';//es parecido al render pero con mas potencial
import { GifGridItem } from '../../components/GifGridItem';


describe('Pruebas en <GifGridItem />', () => {
    const url = 'https://media.tenor.com/images/8f91571e84d8e808fa519529408c20a1/tenor.gif';
    const titulo = 'Hola soy Goku';
    const wrapper = shallow( <GifGridItem title={titulo} url={url} />);
    
    test('Debe de mostrar <GifGridItem /> correctamente', () => {        
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe(titulo);
    })

    test('debe de tener la imagen igual al url y el alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(titulo);
    })
    
    test('debe tener la constante animate__fadeIn ', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__flipInX')).toBe( true );
    })
    

})
