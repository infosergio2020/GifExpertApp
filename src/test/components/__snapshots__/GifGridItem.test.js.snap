// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__flipInX"
>
  <img
    alt="Hola soy Goku"
    src="https://media.tenor.com/images/8f91571e84d8e808fa519529408c20a1/tenor.gif"
  />
  <p>
     
    Hola soy Goku
     
  </p>
</div>
`;
